src/vma/dev/qp_mgr.cpp

```diff
@@ -123,10 +124,53 @@ qp_mgr::~qp_mgr()
 	munmap(m_rq_wqe_idx_to_wrid, m_rx_num_wr * sizeof(*m_rq_wqe_idx_to_wrid));
 	munmap(m_sq_wqe_idx_to_wrid, m_tx_num_wr * sizeof(*m_sq_wqe_idx_to_wrid));
 #endif // DEFINED_VMAPOLL
+
+#ifndef DEFINED_VMAPOLL
+	delete[] m_rq_wqe_idx_to_wrid;
```

Comment:
- On line 78 you have `#ifdef DEFINED_VMAPOLL`. You should use `#else`/`#endif` here.
- What if `!defined(HAVE_INFINIBAND_MLX5_HW_H)`? The `delete` is redundant, though if its like `free()` where you can do `free(NULL)` then you're OK.